samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMessage.java

```diff
@@ -110,67 +105,34 @@ public Object getKey() {
     return key;
   }
 
-  /**
-   * Samza sql relational record. A record consists of list of column values and the associated column names.
-   * A column value could be nested, meaning, it could be another SamzaSqlRelRecord.
-   * Right now we do not store any metadata (like nullability, etc) other than the column name in the SamzaSqlRelRecord.
-   */
-  public static class SamzaSqlRelRecord implements Serializable {
-
-    @JsonProperty("fieldNames")
-    private final List<String> fieldNames;
-    @JsonProperty("fieldValues")
-    private final List<Object> fieldValues;
-
-    /**
-     * Creates a {@link SamzaSqlRelRecord} from the list of relational fields and values.
-     * @param fieldNames Ordered list of field names in the row.
-     * @param fieldValues  Ordered list of all the values in the row. Some of the fields can be null. This could be
-     *                     result of delete change capture event in the stream or because of the result of the outer
-     *                     join or the fields themselves are null in the original stream.
-     */
-    public SamzaSqlRelRecord(@JsonProperty("fieldNames") List<String> fieldNames,
-        @JsonProperty("fieldValues") List<Object> fieldValues) {
-      Validate.isTrue(fieldNames.size() == fieldValues.size(), "Field Names and values are not of same length.");
-
-      this.fieldNames = new ArrayList<>();
-      this.fieldValues = new ArrayList<>();
-
-      this.fieldNames.addAll(fieldNames);
-      this.fieldValues.addAll(fieldValues);
-    }
-
-    /**
-     * Get the field names of all the columns in the relational message.
-     * @return the field names of all columns.
-     */
-    @JsonProperty("fieldNames")
-    public List<String> getFieldNames() {
-      return this.fieldNames;
-    }
-
-    /**
-     * Get the field values of all the columns in the relational message.
-     * @return the field values of all columns.
-     */
-    @JsonProperty("fieldValues")
-    public List<Object> getFieldValues() {
-      return this.fieldValues;
-    }
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((key == null) ? 0 : key.hashCode());
+    result = prime * result + ((samzaSqlRelRecord == null) ? 0 : samzaSqlRelRecord.hashCode());
+    return result;
```

Comment:
would Objects.hashCode or HashCodeBuilder work?

Reply:
Good suggestion. Changed to Objects.hashCode.